src/eos-updater-util.c

```diff
@@ -291,17 +293,25 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
     g_hash_table_insert (hw_descriptors, g_strdup ("product_name"),
                          g_strdup ("EOSUNKNOWN"));
 
-  vendor = g_strdup (g_hash_table_lookup (hw_descriptors, "sys_vendor"));
-  product = g_strdup (g_hash_table_lookup (hw_descriptors, "product_name"));
-  p_id = g_strconcat (cleanstr (vendor), " ", cleanstr (product), NULL);
-  message ("Product group: %s", p_id);
+  *sys_vendor = g_strdup (g_hash_table_lookup (hw_descriptors, "sys_vendor"));
+  *product_name = g_strdup (g_hash_table_lookup (hw_descriptors, "product_name"));
+  return hw_descriptors;
```

Comment:
I wonder if you really need to also return these values here separately, instead of looking them up from the table where you need them